packages/overlay/test/overlay-trigger-extended.test.ts

```diff
@@ -172,92 +172,142 @@ describe('Overlay Trigger - extended', () => {
+                >
+                    <sp-button
+                        id="outer-button"
+                        variant="primary"
+                        slot="trigger"
+                    >
+                        Show Popover
+                    </sp-button>
+                    <sp-popover
+                        id="outer-popover"
+                        slot="click-content"
+                        tip
+                        tabindex="0"
+                        placement="bottom"
+                    >
+                        <sp-dialog no-divider>
+                            This is the overlay content.
+                        </sp-dialog>
+                    </sp-popover>
+                </overlay-trigger>
+                <input
+                    type="text"
+                    id="textfield"
+                    style="position: relative; z-index: 1;"
+                />
+            </div>
+        `);
+
+        // Get a reference to the textfield element
+        const textfield = el.querySelector('#textfield') as HTMLInputElement;
+
+        // Get a reference to the overlay trigger
+        const overlayTrigger = el.querySelector(
+            'overlay-trigger'
+        ) as OverlayTrigger;
+
+        // Get a reference to the button element
+        const button = el.querySelector('sp-button') as Button;
+
+        // Get a reference to the popover element
+        const popover = el.querySelector('sp-popover') as Popover;
+
+        // Wait for the textfield to be connected and rendered
+        await elementUpdated(el);
+
+        // Get a reference to the overlay element that will be opened
         const overlay = overlayTrigger.clickOverlayElement;
-        overlayTrigger.insertAdjacentElement('afterend', textfield);
-
-        // Wait for the textfield to be properly connected and rendered
-        await nextFrame();
 
+        // Verify the overlay is initially closed
         expect(overlay.state, `overlay state`).to.equal('closed');
 
-        // Ensure textfield is visible and focusable
+        // Ensure the textfield is visible and can be focused
         expect(textfield.offsetParent, 'textfield is visible').to.not.be.null;
-        expect(textfield.tabIndex, 'textfield is focusable').to.be.greaterThan(
-            -1
-        );
-
-        // Try multiple approaches to ensure focus works in CI
-        await sendMouseTo(textfield, 'click');
-
-        // If click didn't work, try programmatic focus
-        if (document.activeElement !== textfield) {
-            await textfield.focus();
-        }
+        expect(textfield.tabIndex, 'textfield is focusable').to.be.equal(0);
 
+        // Focus the textfield by clicking it (simulates user interaction)
         await waitUntil(
-            () => document.activeElement === textfield,
-            `clicking focuses textfield (active element is ${document.activeElement?.tagName})`,
-            { timeout: 1000 }
```

Comment:
Without a timeout, these tests will still fail but not give us good error messages to tell us which step failed when it timed out. Instead I recommend increasing the timeout time.